tracing: rename "snowball" tracing to verbose tracing

tbg opened the pull request:
First two commits are #57781

----

Today, we have either

- a Span that was optimized out (noop span)
- a non-recording Span
- a recording Span (snowball span)

In the always-on tracing world (which does not yet exist but should soon), we
have:

- a Span
- a verbose Span

The recording Span from the old world is really the verbose Span in the new
world. This PR realizes this change in naming and removes the word "snowball"
from all tracing-related contexts throughout the codebase.

Release note: None

tbg committed:
tracing: unexport and rename the Snowball baggage item

It didn't need to be exported, and since we will introduce new concepts
as part of #55584, this is a good time to rename the existing concepts
more appropriately, so that the newly added ones will be a natural
extension of what's there.

Ultimately, snowball tracing is a per-trace concept which signifies
"verbose" tracing, i.e. a span that is not a noop span, is recording log
messages, and whose descendants will do so as well.

With always-on tracing the big change we will make is that all spans
will be real spans (i.e. no more noopSpan), and that we will have the
ability to add structured information into these spans, independently
of whether they are "recording". The concept of "recording" will thus
apply only to verbose log messages, and we will rename it accordingly
in a future commit.

Release note: None

tbg committed:
tracing: rename SnowballRecording to RecordingVerbose

With always-on tracing, snowball tracing is really verbose tracing. This
commit makes that naming change on the associated recording mode.

Additionally, since this is the only recording mode that is ever passed
to `StartRecording`, we stop passing it. Instead, we coalesce
`StartRecording(RecordingVerbose)` to `SetVerbose(true)` and
`StopRecording()` to `SetVerbose(false)`. Verbosity thus becomes
something we can toggle on a Span.

Release note: None

tbg committed:
tracing: rename Start{Snowball->Verbose}Trace

Release note: None

tbg committed:
*: "snowball tracing" -> "verbose tracing"

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/57822)
<!-- Reviewable:end -->

knz reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> with nit

> Reviewed 3 of 3 files at r1, 3 of 3 files at r2, 2 of 2 files at r3, 18 of 18 files at r4, 8 of 8 files at r5, 14 of 14 files at r6.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/57822#-:-MOGx24w9Odomr4dl5wr:b-xv64lu)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @irfansharif and @tbg)

___
*[pkg/util/tracing/span.go, line 259 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/57822#-MOGwhJU0l4xGMWx0Y0O:-MOGwhJU0l4xGMWx0Y0P:box65k6) ([raw file](https://github.com/cockroachdb/cockroach/blob/c46f5bb5d99e27bf6a5d95f1844e532224f8cf1e/pkg/util/tracing/span.go#L259)):*
> ```Go
> 	// noop span.
> 	if s.isNoop() {
> 		panic("SetVerbose called on NoopSpan; use the WithForceRealSpan option for StartSpan")
> ```

can you change this to use  `panic(errors.AssertionFailed(...))` that will ensure we get better sentry reporting if it occurs in the wild.


<!-- Sent from Reviewable.io -->

tbg committed:
tracing: add assertion error inside panic

Release note: None

tbg committed:
tracing: clean up pretty-printing of verbose traces

They would previously print a `sb=1` tag, but now that we phased out
the "snowball" nomenclature, this is mostly confusing. Introduce some
light special casing that prevents this tag from being added, and also
remove a (harmless) extra call to setting that tag that had snuck in.

While I was there, I also made these auto-generated tags (the other one
is `unfinished`) more discernible by prefixing them with an underscore
and by giving them an explicit value of `1`.

Release note: None

tbg commented:
Addressed your comment, and added another cleanup commit while I was here.

tbg committed:
tracing: rename IsRecording() to IsVerbose()

Release note: None

tbg commented:
bors r=knz
